apps/web/src/pages/Skills/UpdateUserSkillPage.tsx

```diff
@@ -0,0 +1,663 @@
+import React from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { useIntl } from "react-intl";
+import LightBulbIcon from "@heroicons/react/24/outline/LightBulbIcon";
+import BookmarkSquareIcon from "@heroicons/react/24/outline/BookmarkSquareIcon";
+import PlusCircleIcon from "@heroicons/react/24/solid/PlusCircleIcon";
+
+import {
+  ThrowNotFound,
+  TableOfContents,
+  Pending,
+  Well,
+  Button,
+  Dialog,
+} from "@gc-digital-talent/ui";
+import {
+  Experience,
+  Maybe,
+  Scalars,
+  Skill,
+  SkillLevel,
+  SkillCategory,
+  UserSkill,
+  WhenSkillUsed,
+  useCreateUserSkillMutation,
+  useDeleteUserSkillMutation,
+  useUpdateUserSkillMutation,
+  useUserSkillQuery,
+} from "@gc-digital-talent/graphql";
+import {
+  commonMessages,
+  errorMessages,
+  formMessages,
+  getLocalizedName,
+} from "@gc-digital-talent/i18n";
+import { BasicForm, RadioGroup, enumToOptions } from "@gc-digital-talent/forms";
+import {
+  getTechnicalSkillLevel,
+  getBehaviouralSkillLevel,
+  getTechnicalSkillLevelDefinition,
+  getBehaviouralSkillLevelDefinition,
+} from "@gc-digital-talent/i18n/src/messages/localizedConstants";
+import { notEmpty } from "@gc-digital-talent/helpers";
+import { toast } from "@gc-digital-talent/toast";
+
+import SEO from "~/components/SEO/SEO";
+import Hero from "~/components/Hero/Hero";
+import ExperienceCard from "~/components/ExperienceCard/ExperienceCard";
+import SkillFormDialog from "~/components/SkillFormDialog/SkillFormDialog";
+import useRoutes from "~/hooks/useRoutes";
+
+type PageSection = {
+  id: string;
+  title: React.ReactNode;
+};
+type PageSections = Record<string, PageSection>;
+
+type FormValues = {
+  skillLevel: SkillLevel;
+  whenSkillUsed: WhenSkillUsed;
+};
+
+interface UpdateUserSkillFormProps {
+  userId: Scalars["UUID"];
+  skill: Skill;
+  experiences: Experience[];
+  userSkill?: Maybe<UserSkill>;
+}
+
+const UpdateUserSkillForm = ({
+  userId,
+  experiences,
+  skill,
+  userSkill,
+}: UpdateUserSkillFormProps) => {
+  const intl = useIntl();
+  const paths = useRoutes();
+  const navigate = useNavigate();
+  const [isFormDialogOpen, setFormDialogOpen] = React.useState<boolean>(false);
+  const [currentExperience, setCurrentExperience] =
+    React.useState<Experience | null>(null);
+  const skillName = getLocalizedName(skill.name, intl);
+  const skillDescription = getLocalizedName(skill.description, intl);
+  const hasUserSkill = notEmpty(userSkill);
+  const linkedExperiences = userSkill?.skill?.experiences
+    ?.filter(notEmpty)
+    // Note: Temp filter out unowned experiences
+    .filter((experience) =>
+      experiences.some((exp) => exp.id === experience.id),
+    );
```

Comment:
Oh, I am assuming this is depending on #7688

Reply:
Yeah 😬

Reply:
Now that it has been merged, this was updated accordingly in [update query with elevated experiences](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7686/commits/15152278841c37e683c298f862294682f0ec9fba)